draft-ietf-quic-qpack.md

```diff
@@ -580,40 +581,48 @@ prefix integer.  The remainder of the string literal is unmodified.
 A string literal without a prefix length noted is an 8-bit prefix string literal
 and follows the definitions in [RFC7541] without modification.
 
-## Stream Types
+## Instructions
 
-QPACK instructions occur in three locations, each of which uses a separate
-instruction space:
+There are three separate QPACK instruction spaces. Encoder instructions
+({{encoder-instructions}}) carry table updates, decoder instructions
+({{decoder-instructions}}) carry acknowledgments of table modifications and
+header processing, and header block instructions ({{header-block-instructions}})
+reference the QPACK table state.
 
- - The encoder stream is a unidirectional stream of type `0x48` (ASCII 'H')
-   which carries table updates from encoder to decoder.
+Encoder and decoder instructions appear on the unidirectional stream types
+described in this section. Header block instructions are contained in HEADERS
+and PUSH_PROMISE frames, which are conveyed on request or push streams as
+described in {{HTTP3}}.
 
- - The decoder stream is a unidirectional stream of type `0x68` (ASCII 'h')
-   which carries acknowledgements of table modifications and header processing
-   from decoder to encoder.
+### Encoder and Decoder Streams
 
- - Finally, the contents of HEADERS and PUSH_PROMISE frames on request streams
-   and push streams reference the QPACK table state.
+QPACK defines two unidirectional stream types:
 
-<!-- s/exactly/no more than/  ? -->
-There MUST be exactly one of each unidirectional stream type in each direction.
-Receipt of a second instance of either stream type MUST be treated as a
-connection error of HTTP_WRONG_STREAM_COUNT.  These streams MUST NOT be closed.
-Closure of either unidirectional stream MUST be treated as a connection error of
-type HTTP_CLOSED_CRITICAL_STREAM.
+ - An encoder stream is a unidirectional stream of type `0x48` (ASCII 'H').
+   It carries an unframed sequence of encoder instructions from encoder
+   to decoder.
 
-This section describes the instructions which are possible on each stream type.
+ - A decoder stream is a unidirectional stream of type `0x68` (ASCII 'h').
+   It carries an unframed sequence of decoder instructions from decoder
+   to encoder.
 
-## Encoder Stream {#encoder-stream}
+<!-- s/exactly/no more than/  ? -->
+HTTP/3 endpoints contain a QPACK encoder and decoder. Each endpoint MUST
+initiate a single encoder stream and decoder stream. Receipt of a second
+instance of either stream type MUST treat it as a connection error of type
```

Comment:
Check grammar: "An endpoint that receives a second" or "MUST be treated as"

Reply:
Ack